data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -58,20 +60,22 @@ public class OpenSearchSink extends AbstractSink<Record<Event>> {
   private final OpenSearchSinkConfiguration openSearchSinkConfig;
   private final IndexManagerFactory indexManagerFactory;
   private RestHighLevelClient restHighLevelClient;
-  private IndexManager indexManager;
+  private IndexManager defaultIndexManager;
   private Supplier<AccumulatingBulkRequest> bulkRequestSupplier;
   private BulkRetryStrategy bulkRetryStrategy;
   private final long bulkSize;
   private final IndexType indexType;
   private final String documentIdField;
   private final String routingField;
   private final String action;
+  private String configuredIndexAlias;
 
   private final Timer bulkRequestTimer;
   private final Counter bulkRequestErrorsCounter;
   private final DistributionSummary bulkRequestSizeBytesSummary;
   private OpenSearchClient openSearchClient;
   private ObjectMapper objectMapper;
+  private Map<String, IndexManager> indexManagersMap;
```

Comment:
This should be a cache. I could image users using values here that change over time such that this map would grow indefinitely.

You can look into Guava for some caches in the `com.google.common.cache` package. Also, there is the Caffeine project. But, I don't think Data Prepper current has that in the final class path.

Also, I think this cache should be moved out of this class as part of the refactoring I suggested below.